### PR TITLE
Fix #857  Cannot add user to room.  

### LINKED ISSUE #857: Room can not add user now
Exception while invoking method 'addUserToRoom' TypeError: Cannot read property '_id' of undefined
at _Class.RocketChat.models.Subscriptions._Class.createWithRoomAndUser (/data/rocket.chat/bundle/programs/server/packages/rocketchat_lib.js:947:18)
at [object Object].Meteor.methods.addUserToRoom (/data/rocket.chat/bundle/programs/server/app/server/methods/addUserToRoom.coffee.js:18:37)
at [object Object].methodMap.(anonymous function) (/data/rocket.chat/bundle/programs/server/packages/meteorhacks_kadira.js:2574:30)
at maybeAuditArgumentChecks (/data/rocket.chat/bundle/programs/server/packages/ddp.js:2445:12)
at /data/rocket.chat/bundle/programs/server/packages/ddp.js:1476:20
at [object Object]._.extend.withValue (/data/rocket.chat/bundle/programs/server/packages/meteor.js:989:17)
at /data/rocket.chat/bundle/programs/server/packages/ddp.js:1475:41
at [object Object]._.extend.withValue (/data/rocket.chat/bundle/programs/server/packages/meteor.js:989:17)
at [object Object]._.extend.protocol_handlers.method (/data/rocket.chat/bundle/programs/server/packages/ddp.js:1474:51)
at /data/rocket.chat/bundle/programs/server/packages/ddp.js:1374:43

### PR DESCRIPTION
**Pull request by rwakida**
Fix bug where username incorrectly passed to findOneByUsername.  Also adds argument validation.

### CODE REVIEW

**Comment by Sing-Li** [comment]
@marceloschmidt - insomnia?

**Comment by sampaiodiego** [comment]
@Sing-Li the label is added by waffle.io :stuck_out_tongue:

**Comment by sampaiodiego** [comment]
I'm looking into this.. I found some other bugs, trying to fix them too

**Comment by sampaiodiego** [comment]
btw, thanks @rwakida for the quick fix :smile:

File: server/methods/addUserToRoom.coffee [hunk 1]
```diff
@@ -13,7 +19,7 @@ Meteor.methods
 		if room.usernames.indexOf(data.username) isnt -1
 			return
 
-		newUser = RocketChat.models.Users.findOneByUsername username: data.username
```
Comment (rodrigok) [inline server/methods/addUserToRoom.coffee]: My bar, sorry :(

Comment (rwakida) [inline server/methods/addUserToRoom.coffee]: You made a lot of changes.  It's hard to catch everything.  :smile: 

I like how you consolidated the logic into the models.  It should be a lot easier to maintain.

**Comment by rwakida** [comment]
N/P @sampaiodiego .  I'll be the late night Hawaii tech support.  :grinning:

**Pull request merged**